stylelint-polaris/utils/index.js

```diff
@@ -124,28 +124,6 @@ function testAgainstStringOrRegExp(value, comparison) {
     return comparison.test(value) ? {match: value, pattern: comparison} : false;
   }
 
-  // Check if it's RegExp in a string
-  const firstComparisonChar = comparison[0];
-  const lastComparisonChar = comparison[comparison.length - 1];
-  const secondToLastComparisonChar = comparison[comparison.length - 2];
-
-  const comparisonIsRegex =
-    firstComparisonChar === '/' &&
-    (lastComparisonChar === '/' ||
-      (secondToLastComparisonChar === '/' && lastComparisonChar === 'i'));
-
-  const hasCaseInsensitiveFlag =
-    comparisonIsRegex && lastComparisonChar === 'i';
-
-  // If so, create a new RegExp from it
-  if (comparisonIsRegex) {
-    const valueMatches = hasCaseInsensitiveFlag
-      ? new RegExp(comparison.slice(1, -2), 'i').test(value)
-      : new RegExp(comparison.slice(1, -1)).test(value);
-
-    return valueMatches ? {match: value, pattern: comparison} : false;
-  }
-
```

Comment:
This is a duplicate from what I can tell of the lines above `comparison instanceof RegExp`

Reply:
All of these `utils` were copy/pasted from Stylelint: https://github.com/stylelint/stylelint/blob/main/lib/utils/matchesStringOrRegExp.js#L16

This `matchesStringOrRegExp` util in particular is used as the foundation for many built-in Stylelint rules. Thus we copied it verbatim to expose a familiar interface. Happy for it to be reworked to only allow RegExp literals 👍

Reply:
Awesome context. Thank you @aaronccasanova. Happy to leave as is.